test/test-applications/integrations/TestApplication.ContinuousProfiler/Exporter/SampleNativeFormatParser.cs

```diff
@@ -0,0 +1,308 @@
+    /// <param name="buffer">byte array containing native thread samples format data</param>
+    /// <param name="read">how much of the buffer is actually used</param>
+    internal static List<ThreadSample>? ParseThreadSamples(byte[] buffer, int read)
+    {
+        uint batchThreadIndex = 0;
+        var samples = new List<ThreadSample>();
+        long sampleStartMillis = 0;
+
+        var position = 0;
+
+        // common for samples in a batch
+        var codeDictionary = new Dictionary<int, string>();
+
+        try
+        {
+            while (position < read)
+            {
+                var operationCode = buffer[position];
+                position++;
+                if (operationCode == OpCodes.StartBatch)
+                {
+                    var version = ReadInt(buffer, ref position);
+                    if (version != 1)
+                    {
+                        return null; // not able to parse
+                    }
+
+                    sampleStartMillis = ReadInt64(buffer, ref position);
+
+                    /* if (IsLogLevelDebugEnabled)
+                    {
+                        var sampleStart = new DateTime(
+                            (sampleStartMillis * TimeSpan.TicksPerMillisecond) + TimeConstants.UnixEpochInTicks).ToLocalTime();
+                        Log.Debug(
+                            "Parsing thread samples captured at {date} {time}",
+                            sampleStart.ToLongDateString(),
+                            sampleStart.ToLongTimeString());
+                    }*/
+                }
+                else if (operationCode == OpCodes.StartSample)
+                {
+                    var managedId = ReadInt(buffer, ref position);
+                    var threadName = ReadString(buffer, ref position);
+                    var traceIdHigh = ReadInt64(buffer, ref position);
+                    var traceIdLow = ReadInt64(buffer, ref position);
+                    var spanId = ReadInt64(buffer, ref position);
+
+                    var threadIndex = batchThreadIndex++;
+
+                    var code = ReadShort(buffer, ref position);
+                    if (code == 0)
+                    {
+                        // Empty stack, skip this sample.
+                        continue;
+                    }
+
+                    var threadSample = new ThreadSample
+                    {
+                        Timestamp = new ThreadSample.Time(sampleStartMillis),
+                        TraceIdHigh = traceIdHigh,
+                        TraceIdLow = traceIdLow,
+                        SpanId = spanId,
+                        ManagedId = managedId,
+                        ThreadName = threadName,
+                        ThreadIndex = threadIndex
+                    };
+
+                    ReadStackFrames(code, threadSample, codeDictionary, buffer, ref position);
+
+                    if (threadName == "OpenTelemetry Continuous Profiler Thread")
```

Comment:
magic string, can it take a reference to the existing constant declaration?

Reply:
a2066c9a955f70521c5084e0a546e96c2d7df6aa, please let me know if you are fine with this solution